src/components/PublishContainer/PublishContainer.tsx

```diff
@@ -0,0 +1,140 @@
+import prettyBytes from "pretty-bytes";
+import React, { FunctionComponent, useEffect } from "react";
+import { Redirect } from "react-router-dom";
+import { useConfigContext } from "../../common/context/config";
+import { useFormsContext } from "../../common/context/forms";
+import { usePublishQueue } from "../../common/hook/usePublishQueue";
+import { Config } from "../../types";
+import { Button } from "../../UI/Button";
+import { PublishLoader } from "../../UI/PublishLoader";
+import { SvgIcon, SvgIconCheckCircle } from "../../UI/SvgIcon";
+import { Title } from "../../UI/Title";
+import { Wrapper } from "../../UI/Wrapper";
+import { Container } from "../Container";
+import { NavigationBar } from "../NavigationBar";
+import { ProgressBar } from "../ProgressBar";
+
+interface PublishPage {
+  config: Config;
+}
+
+export const PublishPage: FunctionComponent<PublishPage> = ({ config }) => {
+  const { setConfig } = useConfigContext();
+  const { forms, currentForm, setForms, setActiveFormIndex } = useFormsContext();
+  const { publish, publishState, publishedDocuments } = usePublishQueue(config, forms);
+
+  useEffect(() => {
+    publish();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const documentsPublished = publishState === "CONFIRMED";
+
+  const createAnotherDoc = (): void => {
+    const nextForms = [...forms];
+    nextForms.splice(nextForms.length - 1, 1);
+    setForms(nextForms);
+    setActiveFormIndex(undefined);
+  };
+
+  const iAmDone = (): void => {
+    setConfig(undefined);
+  };
+
+  const getFileSize = (jsonString: string): number => {
+    const m = encodeURIComponent(jsonString).match(/%[89ABab]/g);
+    return jsonString.length + (m ? m.length : 0);
+  };
+
+  if (!config) return <Redirect to="/" />;
+  if (!currentForm) return <Redirect to="/forms-selection" />;
+
+  return (
+    <>
```

Comment:
a bit too much logic in this container, can refactor it into a few individual component for easier testing?

Like the document download part can be one component, the loading part can be one component, etc

Reply:
ok